neon-mmd/websurfx
# 👷 GitHub action to automate release based on semantic versioning

## What does this PR do?

This PR adds a file to automate bumps in release versions each time a PR is successfully closed and merged on Rolling. However if we want to do the same for Stable we can since it would take a small change to the release.yml file. I tested a major release on rolling then I rebased on this branch. Before I had tested minor and patch bumps as well. Anything not following the [semantic-release format](https://github.com/semantic-release/semantic-release) will increment the patch version by one. 

## Why is this change important?

This change is important due to the fact it closes out issue #400 and it stops everyone from having to manually create a release.

## How to test this PR locally?

Place this file in the same location (.github/workflows/release.yml) then successfully merge a PR into the rolling branch. Once this is done you should see the action successfully runs and in a moment you will see the new tag and release. Please create a Personal Access Token or [PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) and assign it these settings below. Not pictured is only setting the token to have control of the websurfx repo (or your fork).
![image](https://github.com/neon-mmd/websurfx/assets/80804614/684dcaf9-c1f9-4e59-8f03-35806959bf16)

Once you are done please copy the token since you will not be able to see it again. Then you must add the PAT as a secret called ADMIN_RIGHTS_TOKEN. Please paste in the token you copied from above here. You may follow these steps https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions?tool=webui#creating-secrets-for-a-repository

Note: I used webui screenshots and notes to be more beginner friendly, but you can do this in the CLI as well.

## Author's checklist

N/A

## Related issues
Closes #400
